### DISCUSSION TITLE
Can't enter setup mode

### DISCUSSION THREAD

**Discussion by ream88** [RavenSystem/esp-homekit-devices#2173] [Q&A]
Ok, I think I'm lost now. In order to get my Sonoff Basic Switch to work with this, I need to do the following, right?

- Flash `fullhaaboot.bin` onto the switch
- Connect to the created Wifi and open http://192.168.4.1:4567
- Configure (or not) my MEPLHAA script
- Configure my Wifi
- Wait for the switch to reboot and get a IP from my DHCP
- Start setup mode by pressing the button for 8 seconds and connect again via the browser at http://10.0.0.34:4567?
- Flash `haamain.bin` onto it?

I'm stuck at the second to last step, as my switch gets an IP but then refuses to show the setup page, whatever I do. Am I doing something wrong? Is the example correct (https://github.com/RavenSystem/esp-homekit-devices/wiki/Switches#Data-History-Characteristics)? Does it include the 8 seconds thingy?

UPDATE:
I was able to open the setup screen once via the new IP, however it was still orange. After adding my MEPLHAA script it stopped again however.

**Comment by duku602** [comment]
Hi, when flashing new device and after setting wifi info, you have to wait ~10 minutes because during this time haa will install the rest haamain etc.

check this page https://github.com/RavenSystem/esp-homekit-devices/wiki/installation#installing-haa 
step 5
